holidays/countries/gibraltar.py

```diff
@@ -0,0 +1,221 @@
+        * [2017](https://web.archive.org/web/20250713073752/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2016s217.pdf)
+        * [2018](https://web.archive.org/web/20250714135250/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2017s224.pdf)
+        * [2019](https://web.archive.org/web/20250713071343/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2018s229.pdf)
+        * [2020](https://web.archive.org/web/20250714135425/https://www.gibraltarlaws.gov.gi/uploads/legislations/banking-and-financial-dealings/B&P%20Holidays/2019s217.pdf)
+        * [2021](https://web.archive.org/web/20250421063152/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2020-5593/download)
+        * [2022](https://web.archive.org/web/20250414134257/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2021-6286/version/21-10-2021/download)
+        * [2023](https://web.archive.org/web/20240831014104/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2022-6735/version/22-02-2023/download)
+        * [2024](https://web.archive.org/web/20250502030403/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2023-7215/version/06-11-2023/download)
+        * [2025](https://web.archive.org/web/20250714135600/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-2024-7456/download)
+    """
+
+    country = "GI"
+    default_language = "en_GB"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # First holiday info available from 2000.
+    start_year = 2000
+    supported_languages = ("en_GB", "en_US")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, GibraltarStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        if self._year >= 2023:
+            # Winter Midterm Bank Holiday.
+            name = tr("Winter Midterm Bank Holiday")
+            winter_midterm_dts = {
+                2024: (FEB, 12),
+            }
+            if dt := winter_midterm_dts.get(self._year):
+                self._add_holiday(name, dt)
+            else:
+                self._add_holiday_3rd_mon_of_feb(name)
+
+        if self._year <= 2022:
+            # Commonwealth Day.
+            name = tr("Commonwealth Day")
+            if self._year >= 2021:
+                self._add_holiday_3rd_mon_of_feb(name)
+            else:
+                self._add_holiday_2nd_mon_of_mar(name)
```

Comment:
AFAIK [it's the same holiday](https://en.wikipedia.org/wiki/Commonwealth_Day#Observance), we can probably simplify the logic up a bit
```suggestion
        name = (
            # Winter Midterm Bank Holiday.
            tr("Winter Midterm Bank Holiday")
            if self._year >= 2023
            # Commonwealth Day.
            else tr("Commonwealth Day")
        )
        winter_midterm_dts = {
            2024: (FEB, 12),
        }
        if dt := winter_midterm_dts.get(self._year):
            self._add_holiday(name, dt)
        elif self._year >= 2021:
            self._add_holiday_3rd_mon_of_feb(name)
        else:
            self._add_holiday_2nd_mon_of_mar(name)
```